### PR TITLE
Add januscript page frontend

### PR DESCRIPTION
**Pull request by ThomasFH1**
### Changes
Use archive as template for januscript page, but without filters and search bar
Add januscript page at /januscript
-

### CODE REVIEW

File: frontend/src/components/pages/januscript/DocumentList.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      alignItems="stretch"
+      sx={(theme) => ({ mb: theme.spacing(8) })}
+    >
+      {documents.map((doc) => (
+        <Grid item md={3} xs={12} sm={6} key={doc.id}>
+          <Card sx={{ height: "100%" }}>
+            <CardActionArea href={doc.webLink ?? ""} sx={{ height: "100%" }}>
+              <CardMedia
+                component="img"
+                image={"https://www.januslinjeforening.no/wp-content/uploads/2016/12/JanuScript_logo.png"}
```
Comment (MagnusHafstad) [inline frontend/src/components/pages/januscript/DocumentList.tsx:29]: Burde være bilde vi har mer kontroll på. Det domene kommer dessuten antakelig bare til å eksistere i noen måneder til.

File: frontend/src/components/pages/januscript/DocumentList.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                }}
+              />
+              <CardContent>
+                <Typography gutterBottom variant="subtitle2">
+                  {doc.title}
+                </Typography>
+                <Typography variant="caption">
+                  {doc.typeDoc
+                    .replace(/_/g, " ")
+                    .replace("ARBOKER", "ÅRBØKER")
```
Comment (MagnusHafstad) [inline frontend/src/components/pages/januscript/DocumentList.tsx:42]: Disse replace funksjonene virker unnødvendige for meg

File: frontend/src/components/pages/januscript/Documents.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  year: number | null;
+};
+
+export const Documents: React.FC<Props> = ({ documentTypes, year }) => {
+  const { refetch, loading, data, error } = useQuery(ArchiveByTypesDocument, {
+    variables: { documentTypes, year },
+    ssr: false,
+  });
+
+  useEffect(() => {
```
Comment (MagnusHafstad) [inline frontend/src/components/pages/januscript/Documents.tsx:20]: Sjekk om dette kan slettes

**Review by MagnusHafstad — CHANGES_REQUESTED** [review]
See comments in code before merging

File: frontend/src/components/pages/januscript/Documents.tsx (hunk 3 above)
Comment (larwaa) [reply frontend/src/components/pages/januscript/Documents.tsx:20]: Yup, kopierer dette fra #1425

This is redundant, and I'll try to outline exactly why.

The way the Apollo Client, and mores specifically, the `useQuery` hook work is roughly something along these lines:


1. Your parent component renders, which has states such as `documentTypes` and `year`, i.e.
```tsx
[documentTypes, setDocumentTypes] = useState(...);
[year, setYear] = useState(...);
```
2. This component is rendered with the current values for the _stateful_ variables `year` and `documentTypes`, for example, this could be `year = 2023` and `documentTypes = ["regnskap"]` 
3. The `useQuery` hook is run with the the current value for `year` and `documentTypes`. The Apollo Client then uses these values in order to generate a _cache key_, which is uses to perform a lookup in the client-side cache. However, no such cache exists as this is the first time this query is run with those values.
4. Since the Apollo Client did not find anything in the client-side cache, it sends the request to our server, which returns list of documents. The Apollo Client then automatically adds the result to the _cache key_ we generated previously, so that any subsequent requests with the same values for `documentTypes` and `year` will hit the cache rather than send a new request to the server.
5. When the user then update the states in the parent component, e.g. `setYear(2022)`, the parent component will re-render, and all child components (importantly, including this component), will re-render. Now, this component will render with `documentTypes = ["regnskap"]` and `year = 2022` rather than `year = 2023`.
6. Same in 3., the `useQuery`-hook is run with the current values for `documentTypes` and `year`, and they're once again used to generate the cache key and we perform a cache-lookup. However, since the result of the previous render had `year = 2023` rather than `year = 2022`, we have no results for the newly generated cache key. As such, we repeat steps 4 and 5, sending a request to the server, storing the results in the new cache key, and rendering the component with the updated documents that match the user's preference.

**TL;DR:** If you update the state of one of the variables in a `useQuery`-hook, the query will be re-run and you'll get the updated results, basically refetching.

I've set up a simple [simple sandbox here to demonstrate this principle on a public API](https://stackblitz.com/edit/nextjs-nmozng?file=pages/index.js)

**Comment by MagnusHafstad** [comment]
Never finished...
